### DISCUSSION TITLE
Group segmentation

### DISCUSSION THREAD

**Discussion by Timpearc** [ultralytics/ultralytics#11248] [General]
![IMG_1164](https://github.com/ultralytics/ultralytics/assets/155924579/1a6bcd6f-4f50-49f3-b711-3dd1e69e01e2)

hi all, 

I’m working on identifying features on a surface, I’ve trained the system on erosion and material loss now I need to identify the holes that are drilled into the surface so they are all groups of the same class holes. 
the holes are in a very specific pattern and it’s not really the actual holes I’m interested in but if they have deformed in shape from the design intent 

I wanted other peoples input I could just label all of them as holes and then decide it’s row number based on position in post processing 

or

I could label all of row 5 holes individually as row5_holes then row6_holes etc

when new all holes are slightly different shapes as show in the representative image how would others attack this problem? 
thanks tim

**Comment by pderrenger** [comment]
@Timpearc hi Tim,

For the scenario you described, labeling each row of holes separately (such as `row5_holes`, `row6_holes`, etc.) sounds like a precise approach if distinguishing between rows is crucial for your analysis, especially since the deformation you're interested in could vary significantly by row. This method allows you to apply specific analyses for each row during post-processing effectively.

If the position within the image consistently correlates with row numbers, using a single label (`holes`) and post-processing to assign row based on position is efficient and simplifies your training process. You can use the bounding box coordinates from the detection to calculate the row dynamically.

Here is a brief example of how you might implement segmentation using YOLOv8 for this, focusing solely on detection, then sorting by position in a post-processing step if you choose the single label approach:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path_to_your_trained_model.pt')

# Predict on an image
results = model('path_to_your_image.jpg')

# Analyze bounding boxes for grouping by rows
for r in results:
    print(r.boxes)  # Assuming you have bounding box info here
```

Deciding on the approach may depend on your specific requirements for accuracy and how much variation exists between rows. Each has its merits!

**Reply by Timpearc** [reply]
Ok thanks for the reply you’ve kind of summed up what I was thinking  I’m first going to try all labeled as one class and see how that goes 
Thanks

**Reply by pderrenger** [reply]
@Timpearc You're welcome! Sounds like a solid plan to start with labeling them as one class and see how it performs. Keep us updated on your progress, and don't hesitate to reach out if you have more questions or need further assistance. Good luck with your project! 👍
